services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -64,12 +66,44 @@ public UUID generateUUID(CreateUpdateTableRequestBody createUpdateTableRequestBo
   public UUID generateUUID(IcebergSnapshotsRequestBody icebergSnapshotsRequestBody) {
     return extractUUIDFromSnapshotJson(
             icebergSnapshotsRequestBody.getJsonSnapshots(),
-            icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getDatabaseId(),
-            icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableId())
+            extractFromTblPropsIfExists(
+                getTableURI(icebergSnapshotsRequestBody),
+                icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableProperties(),
+                DB_RAW_KEY),
+            extractFromTblPropsIfExists(
+                getTableURI(icebergSnapshotsRequestBody),
+                icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableProperties(),
+                TBL_RAW_KEY))
         .orElseGet(
             () -> generateUUID(icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody()));
   }
 
+  /** Simple helper method to obtain tableURI from requestBody. */
+  private String getTableURI(IcebergSnapshotsRequestBody icebergSnapshotsRequestBody) {
+    return icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getDatabaseId()
+        + "."
+        + icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableId();
+  }
+
+  /**
+   * Extracting the value of given key from the table properties map. The main use cases are for
+   * tableId and databaseId where the value captured in tblproperties preserved the casing from
+   * creation. This casing is critical if r/w for this table occurs in a platform with different
+   * casing-preservation contract.
+   */
+  private String extractFromTblPropsIfExists(
+      String tableURI, Map<String, String> tblProps, String rawKey) {
+    if (tblProps == null
```

Comment:
lets make this logic implementation of `extractUUIDFromSnapshotJson( List<String> jsonSnapshots, Map<String, String> tblProps)`

Reply:
IIUC do you mean to say `extractUUIDFromSnapshotJson` should be intaking `IcebergSnapshotsRequestBody` instead of `IcebergSnapshotsRequestBody#getJsonSnapshots()`, so that we would derived `tableURI` from request-body itself ?

Reply:
I thought we could reconstruct the table URI from table Properties. If that's not possible, lets pass the arguments:
`extractUUIDFromSnapshotJson( List<String> jsonSnapshots,dbId, tbId, Map<String, String> tblProps)`

Reply:
I guess I understand what you are asking now. But I found sending in a snapshot-request-body is cleaner.